### PR TITLE
esp32: Extract memory layout definitions to a separate header

### PR DESCRIPTION
**Pull request by yamt**
## Summary
esp32: Extract memory layout definitions to a separate header
## Testing
build tested

### CODE REVIEW

**Comment by Ouss4** [comment]
@yamt could you please squash the two commits?  I think since it's a new file with no code leveraged, it does matter where the license came from.

BTW, I've got a few changes for this layout that I'm about to push.  It changes a bit the layout and introduces a couple of regions (mostly in SMP).

**Comment by yamt** [comment]
> @yamt could you please squash the two commits? I think since it's a new file with no code leveraged, it does matter where the license came from.

i don't think it's appropriate to squash them.

it isn't "a new file with no code leveraged". it's a code extracted from the existing file.
so it should inherit the original copyright notice.

the license change is a logically independent change.
it's in the same PR just because we have a policy to avoid adding a file with non apache license.

> 
> BTW, I've got a few changes for this layout that I'm about to push. It changes a bit the layout and introduces a couple of regions (mostly in SMP).

ok. do you want to make your change in before this one?

**Comment by Ouss4** [comment]
> i don't think it's appropriate to squash them.

Ok.

> ok. do you want to make your change in before this one?

I wanted to, but let's not hold this one.  I'll rebase once merged and update the comments.

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**
